File: 作业7-从无序数组中找出最长的有序长度/java/Solution.java

public class Solution {
	/**
	 * https://leetcode-cn.com/problems/longest-continuous-increasing-subsequence/
	 * @param  nums [description]
	 * @return      [description]
	 */
	public static int findMaxLength(int[] nums) {
		int max = 0, ans = 0;
		for (int i = 0; i < nums.length; i++) {
			if (i > 0 && nums[i - 1] > nums[i]) {
				ans = i;
			}

			max = Math.max(max, i - ans + 1);
		}
		return max;
	}
}

File: 作业7-从无序数组中找出最长的有序长度/php/Solution.php
<?php

class Solution {
	function findMaxLength($nums) {
		$max = 0; $ans = 0;
		for ($i = 0; $i < count($nums);$i++) {
			if ($i > 0; $nums[$i - 1] > $nums[$i]) {
				$ans = $i;
			}
			$max = max($max, $i - $ans + 1);
		}
		return $max;
	}
}